src/Functions/generateRandomStructure.cpp

```diff
@@ -0,0 +1,379 @@
+                writeMapKeyType(rng, buf);
+                writeCString(", ", buf);
+                writeRandomType(column_name, rng, buf, depth + 1);
+                writeChar(')', buf);
+                return;
+            }
+            case TypeIndex::Tuple:
+            {
+                size_t elements = rng() % MAX_TUPLE_ELEMENTS + 1;
+                bool generate_nested = rng() % 2;
+                bool generate_named_tuple = rng() % 2;
+                if (generate_nested)
+                    writeCString("Nested(", buf);
+                else
+                    writeCString("Tuple(", buf);
+
+                for (size_t i = 0; i != elements; ++i)
+                {
+                    if (i != 0)
+                        writeCString(", ", buf);
+
+                    String element_name = "e" + std::to_string(i + 1);
+                    if (generate_named_tuple || generate_nested)
+                    {
+                        writeString(element_name, buf);
+                        writeChar(' ', buf);
+                    }
+                    writeRandomType(element_name, rng, buf, depth + 1);
+                }
+                writeChar(')', buf);
+                return;
+            }
+            default:
+                writeString(magic_enum::enum_name<TypeIndex>(type), buf);
+                return;
+        }
+    }
+
+    void writeMapKeyType(pcg64 & rng, WriteBuffer & buf) const
+    {
+        TypeIndex type = map_key_types[rng() % map_key_types.size()];
+        if (type == TypeIndex::FixedString)
+            writeString("FixedString(" + std::to_string(rng() % MAX_FIXEDSTRING_SIZE_WITHOUT_SUSPICIOUS + 1) + ")", buf);
+        else
+            writeString(magic_enum::enum_name<TypeIndex>(type), buf);
+    }
+
+    void writeLowCardinalityNestedType(pcg64 & rng, WriteBuffer & buf) const
+    {
+        /// Support only String and FixedString (maybe Nullable).
+        String nested_type;
+        bool make_nullable = rng() % 2;
+        if (make_nullable)
+            writeCString("Nullable(", buf);
+
+        if (rng() % 2)
+            writeCString("String", buf);
+        else
+            writeString("FixedString(" + std::to_string(rng() % MAX_FIXEDSTRING_SIZE_WITHOUT_SUSPICIOUS + 1) + ")", buf);
+
+        if (make_nullable)
+            writeChar(')', buf);
+    }
+
+    void writeEnumValues(const String & column_name, pcg64 & rng, WriteBuffer & buf) const
+    {
+        /// Don't generate big enums, because it will lead to really big result
+        /// and slowness of this function, and it can lead to `Max query size exceeded`
+        /// while using this function with generateRandom.
+        ssize_t num_values = rng() % 16 + 1;
+        for (ssize_t i = 0; i != num_values; ++i)
+        {
+            if (i != 0)
+                writeCString(", ", buf);
+            writeString("'" + column_name + "V" + std::to_string(i) + "' = " + std::to_string(i), buf);
+        }
```

Comment:
Would be nice to generate non-consecutive and non-increasing values too, like `'v0' = 42, 'v1' = 13`.

A simple way to generate distinct random numbers is:
```
std::vector<UInt16> values(num_values);
for (UInt16 & x : values)
    x = rng() % (max_value - num_values + 2); // max_value is UINT8_MAX or UINT16_MAX
std::sort(values.begin(), values.end());
for (size_t i = 0; i < num_values; ++i)
    values[i] += i;
std::shuffle(values.begin(), values.end(), rng);
```
(Note that this is biased, in the sense that not all sets of numbers are equally likely, so not good for applications where that matters.)

Reply:
Agree, it would be better. Done.